### PR TITLE
Add max_docs_per_value to diversified sampler documentation

### LINKED ISSUE #5509: [DOC] Diversified Sampler Agg doc does not include max_docs_per_value parameter
**What do you want to do?**
 
- [x] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
The doc https://opensearch.org/docs/latest/aggregations/bucket/diversified-sampler/ is missing the `max_docs_per_value` parameter. Additionally, it would be good to have a short description of what these parameters do. 

For more details see: https://github.com/opensearch-project/OpenSearch/issues/11075

### PR DESCRIPTION
**Pull request by jed326**
### Description
Add the `max_docs_per_value` to the diversified sampler documentation.

### Issues Resolved
Resolves #5509 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _aggregations/bucket/diversified-sampler.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ redirect_from:
 
 # Diversified sampler aggregations
 
-The `diversified_sampler` aggregation lets you reduce the bias in the distribution of the sample pool. You can use the `field` setting to control the maximum number of documents collected on any one shard which shares a common value:
+The `diversified_sampler` aggregation lets you reduce the bias in the distribution of the sample pool by de-duplicating documents with the same `field`. It does so via the `max_docs_per_value` and `field` settings, which limit the maximum number of docs collected on a shard for the provided `field`. Similar to the Sampler aggregation, you can use the `shard_size` setting to control the maximum number of documents collected on any one shard:
```
Comment (vagimeli) [inline _aggregations/bucket/diversified-sampler.md:13]:
```suggestion
The `diversified_sampler` aggregation lets you reduce the bias in the distribution of the sample pool by de-duplicating documents with the same `field`. It does so by using the `max_docs_per_value` and `field` settings, which limit the maximum number of documents collected on a shard for the provided `field`. Similar to the Sampler aggregation, you can use the `shard_size` setting to control the maximum number of documents collected on any one shard:
```

File: _aggregations/bucket/sampler.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ nav_order: 170
 
 # Sampler aggregations
 
-If you're aggregating over millions of documents, you can use a `sampler` aggregation to reduce its scope to a small sample of documents for a faster response. The `sampler` aggregation selects the samples by top-scoring documents.
+If you're aggregating a very large number of documents, you can use a `sampler` aggregation to reduce its scope to a small sample of documents for a faster response. The `sampler` aggregation selects the samples by top-scoring documents.
```
Comment (vagimeli) [inline _aggregations/bucket/sampler.md:11]: What constitutes a "very large number"? This is subjective. We need to give user an idea of what is means for them.

**Review by vagimeli — APPROVED** [review]
Hi @jed326 , Please see the minor copy edits and comment.

File: _aggregations/bucket/sampler.md (hunk 2 above)
Comment (jed326) [reply _aggregations/bucket/sampler.md:11]: "over millions" is also kind of subjective. As a user I'm not sure if this means > 1 million or >10 million or > 100 million. But more importantly there isn't really a threshold here where above x documents the sampler aggregation is useful and below x documents it isn't.

To better illustrate this, there's really 2 (main) use cases for this aggregation type:
1. Some aggregations, like `significant_terms`, still produce relevant results on just a sample of the data so in those cases the sampler aggregation can be used to reduce the cost of running the query.
2. In cases where the query may have a lot of low-quality matches and a few high-quality matches, the sampler aggregation can help focus on just the high-quality matches.

**Review by vagimeli — APPROVED** [review]
@jed326 Do we want to add the use case examples to the documentation?

File: _aggregations/bucket/diversified-sampler.md [hunk 3]
```diff
@@ -8,17 +8,19 @@ redirect_from:
   - /query-dsl/aggregations/bucket/diversified-sampler/
 ---
 
-# Diversified sampler aggregations
+# Diversified sampler
 
-The `diversified_sampler` aggregation lets you reduce the bias in the distribution of the sample pool. You can use the `field` setting to control the maximum number of documents collected on any one shard which shares a common value:
+The `diversified_sampler` aggregation lets you reduce the bias in the distribution of the sample pool by de-duplicating documents with the same `field`. It does so by using the `max_docs_per_value` and `field` settings, which limit the maximum number of documents collected on a shard for the provided `field`. The `max_docs_per_value` setting is an optional parameter used to determine up to how many documents per `field` will be returned. The default value of this setting is `1`.
```
Comment (natebower) [inline _aggregations/bucket/diversified-sampler.md:13]:
```suggestion
The `diversified_sampler` aggregation lets you reduce the bias in the distribution of the sample pool by deduplicating documents containing the same `field`. It does so by using the `max_docs_per_value` and `field` settings, which limit the maximum number of documents collected on a shard for the provided `field`. The `max_docs_per_value` setting is an optional parameter used to determine up to how many documents per `field` will be returned. The default value of this setting is `1`.
```

Comment (natebower) [inline _aggregations/bucket/diversified-sampler.md:13]: Last sentence: "used to determine the maximum number of documents that will be returned per `field`"?

File: _aggregations/bucket/diversified-sampler.md [hunk 4]
```diff
@@ -8,17 +8,19 @@ redirect_from:
...
+
+Similar to the [`sampler` aggregation]({{site.url}}{{site.baseurl}}/aggregations/bucket/sampler/), you can use the `shard_size` setting to control the maximum number of documents collected on any one shard, as shown in the following example:
```
Comment (natebower) [inline _aggregations/bucket/diversified-sampler.md:15]:
```suggestion
Similarly to the [`sampler` aggregation]({{site.url}}{{site.baseurl}}/aggregations/bucket/sampler/), you can use the `shard_size` setting to control the maximum number of documents collected on any one shard, as shown in the following example:
```

File: _aggregations/bucket/sampler.md (hunk 2 above)
Comment (natebower) [inline _aggregations/bucket/sampler.md:11]:
```suggestion
If you're aggregating a very large number of documents, you can use a `sampler` aggregation to reduce its scope to a small sample of documents, resulting in a faster response. The `sampler` aggregation selects the samples by top-scoring documents.
```

Comment (natebower) [inline _aggregations/bucket/sampler.md:11]: Can we replace "its" with the appropriate possessive noun? Or should it just be "the"?

**Review by natebower — COMMENTED** [review]
@vagimeli Please see my comments and changes and let me know if you have any questions. Thanks!

**Pull request merged**